demos/common/python/openvino/model_zoo/model_api/README.md

```diff
@@ -0,0 +1,136 @@
+|----------------------------|--------------------|
+| Background Matting         | <ul><li>`VideoBackgroundMatting`</li><li>`ImageMattingWithBackground`</li></ul> |
+| Classification             | <ul><li>`Classification`</li></ul> |
+| Deblurring                 | <ul><li>`Deblurring`</li></ul> |
+| Human Pose Estimation      | <ul><li>`HpeAssociativeEmbedding`</li><li>`OpenPose`</li></ul> |
+| Instance Segmentation      | <ul><li>`MaskRCNNModel`</li><li>`YolactModel`</li></ul> |
+| Monocular Depth Estimation | <ul><li> `MonoDepthModel`</li></ul> |
+| Named Entity Recognition   | <ul><li>`BertNamedEntityRecognition`</li></ul> |
+| Object Detection           | <ul><li>`CenterNet`</li><li>`DETR`</li><li>`CTPN`</li><li>`FaceBoxes`</li><li>`RetinaFace`</li><li>`RetinaFacePyTorch`</li><li>`SSD`</li><li>`UltraLightweightFaceDetection`</li><li>`YOLO`</li><li>`YoloV3ONNX`</li><li>`YoloV4`</li><li>`YOLOF`</li><li>`YOLOX`</li></ul> |
+| Question Answering         |  <ul><li>`BertQuestionAnswering`</li></ul> |
+| Salient Object Detection   |  <ul><li>`SalientObjectDetectionModel`</li></ul> |
+| Semantic Segmentation      |  <ul><li>`SegmentationModel`</li></ul> |
+
+## Model API Adapters
+
+Model API wrappers are executor-agnostic, meaning it does not implement the specific model inference or model loading, instead it can be used with different executors having the implementation of common interface methods in adapter class respectively.
+
+Currently, `OpenvinoAdapter` and `OVMSAdapter` are supported.
+
+#### OpenVINO Adapter
+
+`OpenvinoAdapter` hides the OpenVINO™ toolkit API, which allows Model API wrappers launching with models represented in Intermediate Representation (IR) format.
+It accepts a path to either `xml` model file or `onnx` model file.
+
+#### OpenVINO Model Server Adapter
+
+`OVMSAdapter` hides the OpenVINO Model Server python client API, which allows Model API wrappers launching with models served by OVMS.
+
+Refer to __[`OVMSAdapter`](adapters/ovms_adapter.md)__ to learn about running demos with OVMS.
+
+For using OpenVINO Model Server Adapter you need to install the package with extra module:
+```sh
+pip install <omz_dir>/demos/common/python[ovms]
+```
+
+## Model API Pipelines
+
+Model API Pipelines represent the high-level wrappers upon the input data and accessing model results management.
+They perform the data submission for model inference, verification of inference status, whether the result is ready or not, and results accessing.
+
+The `AsyncPipeline` is available, which handles the asynchronous execution of a single model.
+
+## Ready-to-use Model API solutions
+
+To apply Model API wrappers in custom applications, learn the provided example of common scenario of how to use Python* Model API.
+
+ In the example, the SSD architecture is used to predict bounding boxes on input image `"sample.png"`. The model execution is produced by `OpenvinoAdapter`, therefore we submit the path to the model's `xml` file.
```

Comment:
@anzhella-pankratova @Wovchena @vladimir-dudnik There could be a confusion with "architecture" definition, especially in documentation. Some people already asked me, why rcnn models in object_detection_demo relate to ssd architecture type. If there any suggestions how to avoid this misunderstandings? Maybe we should rename architecture_type flag in demos and not mention architecture at all, because principle on what we unify models in one wrapper is entirely about task and inputs/outputs structure similarity.

Reply:
Although we refuse to guarantee custom model works, but it should be flexible enough to allow run model which is not from OMZ, but compatible by topology, inputs/outputs with our. The last example, is `yoloface` model, this can work with OMZ demo when properly converted to IR, and that would be excessive to require customer duplicate our yolo wrapper just because model name is different then OMZ model (or even was not set at conversion time).

Reply:
@vladimir-dudnik so what do we decide, for now leave as is?

Reply:
I would answer yes. Imagine you never asked :)